torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1388,7 +1388,8 @@ void ProcessGroupNCCL::heartbeatMonitor() {
   // Case two: desync might be slow or get stuck. Or we get stuck in
   // destructors, we will sleep for some time before calling std::abort() to
   // kill the whole process.
-  if ((terminateProcessGroup_.load() || collectiveDebugInfoMode_.load()) &&
+  if ((terminateProcessGroup_.load() || collectiveDebugInfoMode_.load() ||
```

Comment:
do we even need `collectiveDebugInfoMode_` here at all anymore? is that just for desync_debug feature now, and does it intersect w/ heartbeat sleeping?

Reply:
It's only for desync debug. My guess of the original intention of this variable is: generating the report also takes some time and we don't want monitor thread to kill the process before the report. This logic seems still valid when we turn on desync debug